### PR TITLE
Added documentation file for React components

### PR DESCRIPTION
**Pull request by rupav**
###### Explanation About What Code Achieves:

  - Explanation
solves #31 
Currently, there are many components which make it difficult for a newcomer to parse through, having a documentation of such components will make it easy to know about the components. Hence, I have used an open-source library to generate documentation of components: [react-docgen](https://github.com/reactjs/react-docgen).

Now to generate/modify documentation run from root directory, `yarn run react-dogen`.

###### TODOs:

  - None

### CODE REVIEW

File: docs/Client/componentsDocs.json [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+        "required": false,
+        "description": "",
+        "defaultValue": {
+          "value": "{ }",
+          "computed": false
+        }
+      }
+    }
+  },
+  "candis\\app\\client\\app\\component\\widget\\PipelineEditor.jsx": {
```
Comment (rupav) [inline docs/Client/componentsDocs.json:330]: Added description of `PipelineEditor` component, to show how react-docgen actually generate docs, it just adds them in `description` section.

**Comment by rupav** [comment]
@achillesrasquinha

**Comment by achillesrasquinha** [comment]
Conflict?

**Comment by rupav** [comment]
yeah this being the last PR having rebase issues, will have to make a new PR, just wanted to know if this way is good for React Documentation.

**Comment by rupav** [comment]
Closing this, will be dealt later on.

**Pull request closed without merging**